bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/AbstractExpression.java

```diff
@@ -240,11 +272,36 @@ public Date getFinalFireTime() {
             }
         }
 
-        if (getCandidates().isEmpty()) {
-            return null;
+        Date lastCandidate = null;
+
+        if (!getCandidates().isEmpty()) {
+            if (getCandidates().size() == 1) {
+                lastCandidate = getCandidates().get(0);
+            } else {
+
+                while (getCandidates().size() == maximumCandidates) {
```

Comment:
> Didn't know that cron ends working in the year 2100

It is the to limit the algorithm. Maybe 2100 is even too long, but one has to be optimist and believe that your ESH/OH2 instance can run unattended and without rebooting for the next 84 years. Maybe for performance reasons we can limit that something like "current time" + 1 year, no?